src/validation.cpp

```diff
@@ -1615,22 +1615,27 @@ void static FlushBlockFile(bool fFinalize = false)
     LOCK(cs_LastBlockFile);
 
     CDiskBlockPos posOld(nLastBlockFile, 0);
+    bool flush_failed = false;
 
     FILE *fileOld = OpenBlockFile(posOld);
     if (fileOld) {
         if (fFinalize)
             TruncateFile(fileOld, vinfoBlockFile[nLastBlockFile].nSize);
-        FileCommit(fileOld);
+        flush_failed = FileCommit(fileOld);
         fclose(fileOld);
     }
 
     fileOld = OpenUndoFile(posOld);
     if (fileOld) {
         if (fFinalize)
             TruncateFile(fileOld, vinfoBlockFile[nLastBlockFile].nUndoSize);
-        FileCommit(fileOld);
+        flush_failed = FileCommit(fileOld);
```

Comment:
|=

Reply:
Whoops, and the logic for these is backwards. They should be:
```c++
 flush_failed |= !FileCommit(fileOld);
```

Reply:
Ah yes this should be flush_ok